sql/src/main/java/io/crate/execution/dsl/phases/MergePhase.java

```diff
@@ -59,20 +62,57 @@
     /**
      * expects sorted input and produces sorted output
      */
-    @Nullable private PositionalOrderBy positionalOrderBy;
+    @Nullable
+    private PositionalOrderBy positionalOrderBy;
+
+    public static boolean isNeeded(Collection<String> executionNodes,
+                                   ResultDescription resultDescription,
+                                   boolean isDistributed) {
+        return isDistributed ||
+               resultDescription.hasRemainingLimitOrOffset() ||
+               !resultDescription.nodeIds().equals(executionNodes);
+    }
+
+    public static MergePhase buildForJoin(PlannerContext plannerContext,
```

Comment:
Is this `Join` specific? If so why, and I think if it is we should avoid putting it into `MergePhase`, as the MergePhase should be kept unaware of joins.

Reply:
any suggestion where to move this method?